script/setup.js

```diff
@@ -176,212 +225,306 @@ try {
+								{ context: "package" },
+								{ context: "packages" },
+								{ context: "prettier" },
+								{ context: "spelling" },
+								{ context: "test" },
+							],
+							strict: false,
+						},
+						restrictions: null,
+					}
+				);
+			},
 			{
-				allow_deletions: false,
-				allow_force_pushes: true,
-				allow_fork_pushes: false,
-				allow_fork_syncing: true,
-				block_creations: false,
-				branch: "main",
-				enforce_admins: false,
-				owner,
-				repo: repository,
-				required_conversation_resolution: true,
-				required_linear_history: false,
-				required_pull_request_reviews: null,
-				required_status_checks: {
-					checks: [
-						{ context: "build" },
-						{ context: "compliance" },
-						{ context: "knip" },
-						{ context: "lint" },
-						{ context: "markdown" },
-						{ context: "package" },
-						{ context: "packages" },
-						{ context: "prettier" },
-						{ context: "spelling" },
-						{ context: "test" },
-					],
-					strict: false,
-				},
-				restrictions: null,
+				startText: `Hydrating branch protection settings...`,
+				successText: `Hydrated branch protection settings.`,
+				stopText: `Error hydrating branch protection settings.`,
+				errorText: `Could not hydrate branch protection settings. `,
 			}
 		);
 
-		console.log(chalk.gray`✔️ Done.`);
+		successSpinnerBlock(`Finished API hydration.`);
 	}
 
-	console.log();
-	console.log(chalk.gray`Removing setup script...`);
-
-	await fs.rm("./script", { force: true, recursive: true });
-	await fs.rm(".github/workflows/setup.yml");
-
-	console.log(chalk.gray`✔️ Done.`);
-} catch (error) {
-	console.log(chalk.red(error.stack));
-	caughtError = error;
-} finally {
-	console.log();
-	console.log(
-		chalk.gray`Removing devDependency packages only used for setup...`
+	await pSpinner(
+		async () => {
+			await rm("./script", { force: true, recursive: true });
+			await rm(".github/workflows/setup.yml");
+		},
+		{
+			startText: `Removing setup script...`,
+			successText: `Removed setup script.`,
+			stopText: `Error removing setup script.`,
+			errorText: `Could not remove setup script. `,
+		}
 	);
 
-	try {
-		await $`pnpm remove @clack/prompts all-contributors-cli chalk octokit npm-user replace-in-file title-case -D`;
-	} catch (error) {
-		console.log("Error uninstalling packages:", error);
-		caughtError = error;
-	}
-
-	console.log(chalk.gray`✔️ Done.`);
-}
+	await pSpinner(
+		async () => {
+			await $`pnpm remove @clack/prompts all-contributors-cli chalk octokit npm-user replace-in-file title-case -D`;
+		},
+		{
+			startText: `Removing devDependency packages only used for setup...`,
+			successText: `Removed devDependency packages only used for setup.`,
+			stopText: `Error removing devDependency packages only used for setup.`,
+			errorText: `Could not remove devDependency packages only used for setup. `,
+		}
+	);
 
-console.log();
+	outro(chalk.green`Great, looks like everything worked! 🎉`);
```

Comment:
[Nit] The intro is in blue and this is in green... what do you think about making both blue, so they match up?

```suggestion
	outro(chalk.blue`Great, looks like everything worked! 🎉`);
```

Reply:
Right that would be great !